### DISCUSSION TITLE
Add Embedded Terminal in File Manager

### DISCUSSION THREAD

**Discussion by risalfajar** [linuxdeepin/developer-center#2910] [Features Request & Ideas | 特性请求 & 头脑风暴]
They have this in Nautilus and Dolphin, hope Deepin File Manager can have this too.

It's very useful and safer than using "open as root" directly.

![480px-Dolphin_terminal.png](https://user-images.githubusercontent.com/28381124/87235644-64e91200-c408-11ea-832a-90643982e271.png)

**Comment by XMuli** [comment]
This has already been discussed, see linuxdeepin/developer-center#1770.

**Comment by risalfajar** [comment]
> This has already been discussed, see linuxdeepin/developer-center#1770.

No, it's different. This is an embedded terminal, means a terminal that exist inside the file manager. What you mentioned is open file manager as root, which is totally different.

**Comment by XMuli** [comment]
Thank you. This suggestion will be seen by the  dde-file-manager team.

**Comment by ghost** [comment]
> It's very useful and safer than using "open as root" directly.

It would be more useful to have a global hotkey to open the Terminal's quake window IMO.

**Comment by risalfajar** [comment]
> > It's very useful and safer than using "open as root" directly.
> 
> It would be more useful to have a global hotkey to open the Terminal's quake window IMO.

Is quake a drop-down terminal? If yes then it's different with embedded terminal in file manager, because embedded file manager terminal follows which directory is opened in the file manager. It's opened in the folder which is displayed in the current view.

**Comment by ghost** [comment]
> Is quake a drop-down terminal?

Yes, only [it actually doesn't drop-down](https://github.com/linuxdeepin/deepin-terminal/issues/7) currently. But that's unrelated. The hotkey for it I mentioned in the linked issue.

Would it be reasonable if there was an option to pop that quake window from the currently active location within file manager and anchor it to the active view somehow? This would simplify things quite a bit from a coding perspective and might still meet your needs. Otherwise maybe a tool like [ranger](http://ranger.github.io/) could help you get out of GUI-land entirely.

**Comment by deepin-CD** [comment]
Thanks for your feedback.
